### ISSUE TITLE
V2 API General Feedback

### ISSUE THREAD

**Issue by cloneofghosts** [Pirate-Weather/pirateweather#155]
### Describe the issue

I'm sure you already have a list of things to fix/implement on the V2 Alpha endpoint but here's a list of things that I noticed:

- [ ] negative Longitudes displaying incorrectly (-75 shows as 285)
- [x] currently Block Missing
- [x] daily Block Missing
- [x] alerts Block Missing
- [x] Summary/Icon data points missing
- [x] `precipType` seems broken
- [x] hourly data points not in the same order as the live API
- [x] units parameter not working correctly
- [ ] visibility is not capped
- [x] some data points using incorrect units for SI units (pressure, visibility, cloudCover, humidity and precipProbability)
- [x] nearestStormDistance and nearestStormBearing shouldn't be on the hourly block
- [x] sourceTimes is a floating point number instead of an integter
- [x] many data points being negative when it doesn't make sense (uvIndex, nearestStormDistance, etc.)
- [x] `moonPhase` and daily `precipAccumulation` aren't rounded to two decimal places
- [ ] currently, hourly and daily precipitation data points are rounded to two decimal places instead of four
- [ ] `windBearing` and `nearestStormBearing` is not a whole number
- [ ] precipIntensityError seems to always display 2 on the minutely block
- [x] `time` parameter on hourly block is a floating point number
- [ ] hourly/minutely block seems to show array positioning
- [ ] minutely and currently precipitation data seems to be using GFS/GEFS in the CONUS domain
- [ ] partly cloudy always shows night icon in hourly and currently data
- [ ] temperature/apparent temperature/dew point cannot be below zero when they should be able to
- [ ] pressure value seems broken on hourly data
- [ ] most time data points are show the incorrect time on daily data points
- [ ] `temperatureLow` is showing this mornings low instead of the low for tonight (could be an issue for all days but the low time is broken)
- [ ] Precipitation Probability can go over 1
- [ ] currently data sometimes shows a clear icon with no day/night suffix
- [ ] currently `precipType` shows clear instead of none
- [ ] daily `temperatureHigh` doesn't seem to work correctly. Most cases it shows a temperature lower than the `temperatureMax` when they should be around the same temperature

If anyone else is using the V2 Alpha API feel free to add things that you've run into. Also for things that are fixed on the V2 Alpha endpoint already do you want to close the issue or leave it open until it launches? I'm fine with leaving it open but it would clean up the issue list and the project list if marked as complete.

### Acknowledgements

- [X] I have searched this repository and [Home Assistant Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have read through the [API documentation](https://pirateweather.net/en/latest/API/) before opening this issue.
- [X] I have written an informative title.

**Comment by cloneofghosts** [comment]
@alexander0042 I'm guessing this is something you're working on? Been getting a Internal Server Error message whenever I try to visit the V2 endpoint URL and the V2 dev URL still does not work.

**Comment by alexander0042** [comment]
You guessed right- I was pulling together a paper for a conference last month, and then away skiing (and checking the weather constantly) this week. Back at this next week, so should be able to start checking some of these off! I really appreciate you putting this list together, since I'll let me keep track of the main things ✅

**Comment by cloneofghosts** [comment]
@alexander0042 The V2 Alpha API has been returning an Internal Server Error for the past day or so. Did you push up an update which broke things or did it just happen to break by itself?

**Comment by cloneofghosts** [comment]
Has the NBM data stopped integrating? I thought the model ran every hour and under sourceTimes I see this `"nbm": "2024-03-05 17Z",`

I thought that the API would fall-back to use other sources if the one used in the forecasts hasn't been updated in x number of hours. From what I can tell the data is still somewhat accurate but the temperatures are way off.

**Comment by cloneofghosts** [comment]
@alexander0042 I was getting an internal server error this morning and then it worked for a short while. Now I'm getting:

This site can’t be reachedpiratev2lb-a90c79daaddc2625.elb.us-east-1.amazonaws.com took too long to respond.
Try:

Checking the connection
[Checking the proxy and the firewall](chrome-error://chromewebdata/#buttons)
ERR_TIMED_OUT

You working on something and it broke the site or is something else going on? The dev endpoint seems to be broken as well.

**Comment by alexander0042** [comment]
Yea, there was a bit of chaos there while I tried a bunch of different things. Adding NBM in turned out to be a bit more complex than I expected, since it meant that which model was being used to grab a specific variable became much more dynamic, instead of just HRRR/ GFS. But it's all in there now, and crucially, the infrastructure to actually serve data from it is now in place, which was a key step. I also ended up focusing more on response speed than I initially planned, since the times were pretty good, I wanted to see how fast I could get them (answer- real quick, at least 10x the speed of the old one, sometimes closer to 50x!)

If it makes sense to you, I'm going to start a new issue thread and pin it with some notes on version 2. I can't say enough how great your details are in this one, and I'll go through them one by one now that all the infrastructure is in place, but my ideas is adding some notes for others to a top level comment (or maybe as a discussion)?
